### DISCUSSION TITLE
How to compress more or is there a best way to save canvas.toJSON

### DISCUSSION THREAD

**Discussion by alensprout** [fabricjs/fabric.js#9796] [Q&A]
I am creating a custom product designer and Im using fabric js and users supposed to be able upload images locally to the canvas and lets say they will save their work I used canvas.toJSON to save it but as expected it's size is high since users can upload images in canvas and images become datauri by default is there a way to compress it more?

**Comment by asturur** [comment]
If the save is meant for a later sessions and you are  using dataUri i don't see a workaround.
You should upload the images somewhere. I don't see a way around.
